x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeSourceHandler.java

```diff
@@ -89,20 +124,6 @@ public int bufferSize() {
         }
     }
 
-    public void addCompletionListener(ActionListener<Void> listener) {
```

Comment:
Moved this to the constructor to enforce that callers register to receive errors.